File: ueueo-framework/ueueo-domain-driven-desgin/src/main/java/com/ueueo/ddd/application/contracts/dtos/ExtensibleCreationAuditedEntityDto.cs
using System;
using Volo.Abp.Auditing;
using Volo.Abp.Data;

namespace Volo.Abp.Application.Dtos;

/**
 * This class can be inherited by DTO classes to implement <see cref="ICreationAuditedObject"/> interface.
 * It also implements the <see cref="IHasExtraProperties"/> interface.
*/
 * <typeparam name="TPrimaryKey">Type of primary key</typeparam>
[Serializable]
public abstract class ExtensibleCreationAuditedEntityDto<TPrimaryKey> : ExtensibleEntityDto<TPrimaryKey>, ICreationAuditedObject
{
     * <inheritdoc />
    public DateTime CreationTime;// { get; set; }

     * <inheritdoc />
    public ID CreatorId;// { get; set; }

    protected ExtensibleCreationAuditedEntityDto()
        : this(true)
    {

    }

    protected ExtensibleCreationAuditedEntityDto(boolean setDefaultsForExtraProperties)
        : base(setDefaultsForExtraProperties)
    {

    }
}

/**
 * This class can be inherited by DTO classes to implement <see cref="ICreationAuditedObject"/> interface.
 * It also implements the <see cref="IHasExtraProperties"/> interface.
*/
[Serializable]
public abstract class ExtensibleCreationAuditedEntityDto : ExtensibleEntityDto, ICreationAuditedObject
{
     * <inheritdoc />
    public DateTime CreationTime;// { get; set; }

     * <inheritdoc />
    public ID CreatorId;// { get; set; }

    protected ExtensibleCreationAuditedEntityDto()
        : this(true)
    {

    }

    protected ExtensibleCreationAuditedEntityDto(boolean setDefaultsForExtraProperties)
        : base(setDefaultsForExtraProperties)
    {

    }
}

File: ueueo-frameworks/ueueo-ddd-application/src/main/java/com/ueueo/ddd/application/contracts/dtos/ExtensibleCreationAuditedEntityDto.java
package com.ueueo.ddd.application.contracts.dtos;

import com.ueueo.ID;
import com.ueueo.auditing.ICreationAuditedObject;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.Date;

/**
 * This class can be inherited by DTO classes to implement <see cref="ICreationAuditedObject"/> interface.
 * It also implements the <see cref="IHasExtraProperties"/> interface.
 */
@EqualsAndHashCode(callSuper = true)
@Data
public abstract class ExtensibleCreationAuditedEntityDto extends ExtensibleEntityDto implements ICreationAuditedObject {

    private Date creationTime;

    private ID creatorId;

    protected ExtensibleCreationAuditedEntityDto() {
        this(true);
    }

    protected ExtensibleCreationAuditedEntityDto(boolean setDefaultsForExtraProperties) {
        super(setDefaultsForExtraProperties);
    }
}
